rfcs/text/0044-fileless-execution-linux.md

```diff
@@ -0,0 +1,132 @@
+# 0044: Fileless execution on Linux
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **2024-03-20** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+This RFC proposes adding new fields and event types to enhance the detection of fileless malware execution and related malicious activities on Linux systems.
+
+The new fields include:
+ * file.is_memfd - Indicates if the file is an anonymous file descriptor (memfd) created using the memfd_create system call.
+ * file.is_shmem - Indicates if the file is a POSIX shared memory object created using the shm_open system call, typically located in /dev/shm.
+ * process.is_setuid - Indicates if the process has the setuid bit set, allowing it to run with the privileges of its owner.
+ * process.is_setgid - Indicates if the process has the setgid bit set, allowing it to run with the privileges of its group.
+ * process.is_memfd - Indicates if the process was executed from a memory file descriptor (memfd).
+ * process.inode_nlink - Number of links to the inode of the process executable file, obtained from the i_nlink variable in the inode structure.
```

Comment:
A "process" and the "processes's executable file" are different things, and I think this might be mixing them too much. This could be renamed to make the separation more clear, but that's something to discuss in later stages.

Reply:
Yes, the `inode_nlink` can be a useful signal to detect anonymous file execution but it is quite low level and less transparent than the other ones. We could discuss in the next stage if we should rephrase it or just remove it for simplicity and rely on `is_memfd `as the only signal.